### PR TITLE
[Keyboard] Add Bacca70 keyboard

### PR DESCRIPTION
**Pull request by thompson-ele**
## Description
Added the Bacca70 keyboard to QMK including default and VIA keymaps

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bacca70/config.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/bacca70/config.h:66]:
```suggestion
```

File: keyboards/bacca70/info.json [hunk 2]
```diff
@@ -0,0 +1,172 @@
+{
+    "keyboard_name": "Bacca70",
+    "url": "https://3dkeebs.com/",
+    "maintainer": "keebnewb",
+    "width": 16,
+    "height": 5,
```
Comment (drashna) [inline keyboards/bacca70/info.json:6]:
```suggestion
```

File: keyboards/bacca70/rules.mk [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = STM32F072
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/bacca70/rules.mk:7]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/bacca70/rules.mk [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/bacca70/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bacca70/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Bacca70
+
+![Bacca70](https://imgur.com/TQMALjw.png)
```
Comment (fauxpark) [inline keyboards/bacca70/readme.md:3]:
```suggestion
![Bacca70](https://imgur.com/TQMALjwh.png)
```

File: keyboards/bacca70/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = STM32F072
```
Comment (fauxpark) [inline keyboards/bacca70/rules.mk:2]:
```suggestion
MCU = STM32F072

# Bootloader selection
BOOTLOADER = stm32-dfu
```

File: keyboards/bacca70/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/bacca70/keymaps/via/keymap.c:21]: This needs a bit of cleaning up, there are mixed tabs and spaces.

File: keyboards/bacca70/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/bacca70/keymaps/default/keymap.c:21]: As above.

File: keyboards/bacca70/keymaps/dede-special/keymap.c [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+// mod tap bracket shifts
+#define LBR_SFT MT(MOD_LSFT, KC_LBRC)
+#define RBR_SFT MT(MOD_RSFT, KC_RBRC)
+
+// Arrow/TenKey/Media Keys/Reset hold with Backspace tap
+#define BSPC_LR LT(3, KC_BSPC)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] = LAYOUT_debaccabean(
```
Comment (fauxpark) [inline keyboards/bacca70/keymaps/dede-special/keymap.c:38]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/bacca70/keymaps/via/keymap.c (hunk 7 above)
Comment (thompson-ele) [reply keyboards/bacca70/keymaps/via/keymap.c:21]: @fauxpark I've updated the keymaps and changed the spaces to tabs 👍

Comment (fauxpark) [reply keyboards/bacca70/keymaps/via/keymap.c:21]: I think I'd prefer spaces, tbh:
image

Comment (thompson-ele) [reply keyboards/bacca70/keymaps/via/keymap.c]: @fauxpark updated! sorry I switched to a different code editor and that helped 😂

**Review by fauxpark — APPROVED** [review]

File: keyboards/bacca70/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+![Bacca70](https://imgur.com/TQMALjwh.png)
+
+The Bacca70 is a winkeyless (WKL) 60% layout that includes a F-row. It's classy, hefty and it was designed by Chewwy (of MechsOnDeck) and BlindAssassin.
+
+* Keyboard Maintainer: [keebnewb](https://github.com/thompson-ele)  
+* Hardware Supported: Bacca70 PCB and the ortholinear variant the DeBaccaBean PCB
+* Hardware Availability: [3DKeebs](https://3dkeebs.com/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/bacca70/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

Comment (thompson-ele) [reply keyboards/bacca70/readme.md]: @drashna updated 👍

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**